### PR TITLE
 HaskellLintBear.py: Import CabalRequirement

### PR DESCRIPTION
**Pull request by ankit01ojha**
Closed https://github.com/coala/coala-bears/issues/1358

### CODE REVIEW

**Comment by sims1253** [comment]
We use Closes instead of Closed. Could you change that?
Use `git commit --amend` and change the commit message. Then do `git push -f`to force a push and overwrite the old commit

File: bears/haskell/HaskellLintBear.py [hunk 1]
```diff
@@ -19,7 +21,9 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {CabalRequirement(
+        package='ghc-mod', version='5.6.0'),
```
Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: `package='ghc-mod'` is not correct.

Comment (Techievena) [inline bears/haskell/HaskellLintBear.py]: For format of using CabalRequirement refer doctests of https://gitlab.com/coala/package_manager/blob/master/dependency_management/requirements/CabalRequirement.py

**Comment by ankit01ojha** [comment]
@Techievena Should I write CabalRequirements in place of ghc-mod and I don't know what should I write in version.

**Comment by jayvdb** [comment]
@ankit01ojha , first understand what `CabalRequirement` is.  use cabal on your own computer.  If you cant use cabal, asked to be unassigned from the issue.

**Comment by ankit01ojha** [comment]
modified!

File: bears/haskell/HaskellLintBear.py [hunk 2]
```diff
@@ -19,7 +21,9 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {CabalRequirement(
+        package='hlint', version='1.9.35-1'),
```
Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: `1.9.35-1` isnt useful. that is an apt version specification
`1.9.35` is the correct version specifier for `cabal install`

Comment (Techievena) [inline bears/haskell/HaskellLintBear.py]: Also no need to mention `package=` and `version=`

Comment (Techievena) [inline bears/haskell/HaskellLintBear.py]: Also @jayvdb do we need DistributionRequirement? We are installing hlint using cabal means shouldn't DistributionRequirement be deprecated?

**Comment by jayvdb** [comment]
ack 8b207d7

**Comment by jayvdb** [comment]
needs rebase

**Comment by sils** [comment]
I object to this PR as per https://github.com/coala/coala-bears/issues/1359#issuecomment-277489451

**Comment by ankit01ojha** [comment]
@sils and @jayvdb any corrections that should be done in this?

**Comment by jayvdb** [comment]
blocked on discussion about requirements on the issue

**Comment by sils** [comment]
and not approved since we seem to agree that this would introduce bugs if implemented that way

File: bears/haskell/HaskellLintBear.py [hunk 3]
```diff
@@ -19,7 +21,8 @@ class HaskellLintBear:
     """
 
     LANGUAGES = {'Haskell'}
-    REQUIREMENTS = {DistributionRequirement(apt_get='hlint')}
+    REQUIREMENTS = {CabalRequirement(package='hlint', version='1.9.35'),
```
Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: You can now use [AnyOneOfRequirements](https://gitlab.com/coala/package_manager/blob/master/dependency_management/requirements/AnyOneOfRequirements.py) to wrap these two requirements in an OR statement.

Comment (jayvdb) [inline bears/haskell/HaskellLintBear.py]: See also https://github.com/coala/coala-bears/issues/1119

**Comment by ankit01ojha** [comment]
@jayvdb , sorry but I have created a new PR #2195 . And I am closing this one.

**Pull request closed without merging**